src/service/xhr-impl.js

```diff
@@ -94,6 +96,15 @@ export class Xhr {
   constructor(win) {
     /** @const {!Window} */
     this.win = win;
+
+    // TODO(@jridgewell, #9048): Once we figure out if ampdoc can always be
+    // found, remove the try catch.
+    try {
+      this.ampdoc_ = ampdocServiceFor(win).getAmpDoc();
+    } catch (e) {
+      this.ampdoc_ = null;
+      dev().error('XHR', e, 'XHR service failed to find ampdoc');
```

Comment:
The probability of failure in PWA case here is about 100% :).

I think we just need to restrict this reporting/warning to single-doc mode. The real issue is only in this mode anyway. See `AmpDocService.isSingleDoc()`